File: skolo-ui/src/app/_interfaces/subject/SubjectDto.ts
export interface SubjectDto {
    id: string; // Use string for GUIDs in TypeScript
    subjectName: string;
    class: string;
    price: number
    teacherId: string;
    subjectId: string;
    teacherFirstName: string;
    teacherLastName: string;
    contents: ContentDto[] |any;
    enrollItemsCount:any;
    teacherRating: number
  }
  export interface ContentDto {
    id: string; 
    contentName: string;
    subjectId: string;
  }

File: SkoloInstituteAPI/SkoloInstitute/Entities/DataTransferObjects/Subject/SubjectDto.cs
using SkoloInstitute.Entities.DataTransferObjects.Grade;

namespace SkoloInstitute.Entities.DataTransferObjects.Subject
{
    public class SubjectDto
    {
        public Guid Id { get; set; }
        public string SubjectName { get; set; }
        public string Class { get; set; }
        public decimal Price { get; set; }
        public Guid TeacherId { get; set; }
        public string TeacherFirstName { get; set; }
        public string TeacherLastName { get; set; }
        public int TeacherRating { get; set; }
        public IEnumerable<ContentDto> Contents { get; set; }
        public int EnrollItemsCount { get; set; }

    }
}
